### PR TITLE
REFACTOR-#5718: add `columns` parameter for `get_dtypes` function

### LINKED ISSUE #5718: REFACTOR: add `columns` parameter for `get_dtypes` function
This will reduce code duplication in read functions.

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5718 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/core/io/text/excel_dispatcher.py [hunk 1]
```diff
@@ -221,18 +221,14 @@ def _read(cls, io, **kwargs):
...
         # over the whole column for each column. The index is set below.
-        dtypes = cls.get_dtypes(dtypes_ids)
+        dtypes = cls.get_dtypes(dtypes_ids, column_names)
 
-        data_ids = cls.build_partition(data_ids, row_lengths, column_widths)
-        # Set the index for the dtypes to the column names
-        if isinstance(dtypes, pandas.Series):
-            dtypes.index = column_names
-        else:
-            dtypes = pandas.Series(dtypes, index=column_names)
```
Comment (anmyachev) [inline modin/core/io/text/excel_dispatcher.py:235]: This code is often duplicated for read functions, it is now in `get_dtypes` function. This is the main change.

File: modin/core/io/text/csv_glob_dispatcher.py [hunk 2]
```diff
@@ -231,18 +231,14 @@ def _read(cls, filepath_or_buffer, **kwargs):
...
             new_index.name = pd_df_metadata.index.name
 
+        partition_ids = cls.build_partition(partition_ids, row_lengths, column_widths)
+
         # Compute dtypes by getting collecting and combining all of the partitions. The
         # reported dtypes from differing rows can be different based on the inference in
         # the limited data seen by each worker. We use pandas to compute the exact dtype
         # over the whole column for each column. The index is set below.
-        dtypes = cls.get_dtypes(dtypes_ids) if len(dtypes_ids) > 0 else None
+        dtypes = cls.get_dtypes(dtypes_ids, column_names)
```
Comment (anmyachev) [inline modin/core/io/text/csv_glob_dispatcher.py:240]: This function is synchronous, so it is better to execute it after `build_partition`.

File: modin/core/storage_formats/pandas/parsers.py [hunk 3]
```diff
@@ -227,6 +230,8 @@ def get_dtypes(cls, dtypes_ids):
             index and types of columns are used as values for full resulting
             frame.
         """
+        if len(dtypes_ids) == 0:
+            return None
```
Comment (dchigarev) [inline modin/core/storage_formats/pandas/parsers.py:234]: shouldn't we just return an empty series?

**Review by dchigarev — COMMENTED** [review]
lgtm! left one comment

File: modin/core/storage_formats/pandas/parsers.py (hunk 3 above)
Comment (anmyachev) [reply modin/core/storage_formats/pandas/parsers.py:234]: This value is then passed to `PandasDataframe` inside `dtypes` parameter which default values is None. So I think we shouldn't.

Comment (anmyachev) [reply modin/core/storage_formats/pandas/parsers.py:234]: I will update the docstring.

**Review by dchigarev — APPROVED** [review]

**Pull request merged**